### PR TITLE
[Prototype][DO NOT MERGE]Remove Arena's Gr00t Docker; use Isaac-Gr00t clients

### PR DESCRIPTION
**Pull request by cvolkcvolk**
## Summary [Draft; Do not merge]

This is a first shot at removing the Gr00t Docker and its dependecies. The Gr00t model is run through the official provided Isaac-Gr00t policy-server implementation.

- Remove Arena's duplicated `remote_policy/` module that re-implemented Isaac-GR00T's `server_client.py` (ZMQ+msgpack server, client, serializer)
 - Remove `Dockerfile.gr00t_server` and `run_gr00t_server.sh` (GR00T repo provides its own server)
 - Add `Gr00tRemoteClosedloopPolicy` — a thin wrapper that uses GR00T's native `PolicyClient` and reuses the same obs/action translation pipeline as the local `Gr00tClosedloopPolicy`
- Remove `ClientSidePolicy`, `ActionChunkingClientSidePolicy`, and `Gr00tRemoteServerSidePolicy` wrapper classes

Arena no longer maintains any Gr00t networking code. Each policy framework (GR00T, OpenPI, etc.) owns its own server and client.

## Motivation

Arena's `remote_policy/` module duplicated Isaac-GR00T's `gr00t/policy/server_client.py` — same ZMQ REP/REQ pattern, same msgpack serializer, same endpoint dispatch. Every protocol change in GR00T required a matching change in Arena. Meanwhile, other frameworks use entirely different protocols (OpenPI uses WebSocket, OpenVLA uses HTTP REST), making Arena's single-protocol approach a dead end for "bring any policy" evaluation.

The observation and action translation (`gr00t_core.py`) is the hard, valuable part and it already exists and is kept.

## What stays the same

- `PolicyBase` ABC (minus `is_remote`)
- `ActionChunkingState` (multi-env GPU chunk buffer)
- `gr00t_core.py` (obs/action translation pipeline)
- `Gr00tClosedloopPolicy` (local, in-process)
- `policy_runner.py` / `eval_runner.py` evaluation loop

## Example 1: DROID Pick-and-Place (using pre-trained HuggingFace model)

The policy model runs in GR00T's own Docker container, and Arena runs in its base Docker container.

You will need **two terminals**.

### Terminal 1: Start the GR00T Policy Server

This starts the official **Isaac-GR00T Docker container** and launches the policy server inside it. The server loads the model and listens for inference requests over ZMQ.

**Build the GR00T Docker image (first time only):**

```bash
cd submodules/Isaac-GR00T
bash docker/build.sh
```

This builds a Docker image called `gr00t-dev` based on `nvcr.io/nvidia/pytorch:25.04-py3` with all GR00T dependencies installed.

**Launch the policy server:**

```bash
docker run -it --rm --gpus all \
  -p 5555:5555 \
  gr00t-dev \
  bash -c "cd /workspace/gr00t && uv run python gr00t/eval/run_gr00t_server.py \
    --model-path nvidia/GR00T-N1.6-DROID \
    --embodiment-tag OXE_DROID \
    --device cuda \
    --host 0.0.0.0 --port 5555"
```

Wait until you see: `Server is ready and listening on tcp://0.0.0.0:5555`

**Find the server IP** — inside the GR00T container, run:

```bash
hostname -I
```
Note this IP (e.g., `192.168.222.2`). You will use it in Terminal 2.

### Terminal 2: Run Arena Evaluation

This starts the **Arena base Docker container** (Isaac Sim + Arena, no GR00T model dependencies) and runs the evaluation against the remote policy server.

**Start the Arena Docker container:**

```bash
./docker/run_docker.sh
```

**Run the evaluation** inside the Arena container:

```bash
/isaac-sim/python.sh isaaclab_arena/evaluation/policy_runner.py \
  --viz kit \
  --policy_type isaaclab_arena_gr00t.policy.gr00t_remote_closedloop_policy.Gr00tRemoteClosedloopPolicy \
  --policy_config_yaml_path isaaclab_arena_gr00t/policy/config/droid_manip_gr00t_closedloop_config.yaml \
  --remote_host <GROOT_SERVER_IP> --remote_port 5555 \
  --enable_cameras --num_episodes 5 \
  pick_and_place_maple_table --embodiment droid_abs_joint_pos
```

Replace `<GROOT_SERVER_IP>` with the IP from Terminal 1 (e.g., `192.168.222.2`).

---

## Example 2: G1 Loco-Manipulation Pick-and-Place (using Arena's fine-tuned model still works)

Same two-terminal setup, but using Arena's fine-tuned G1 model from HuggingFace.

#### Prerequisites: Download the model (one time, on the host). For some reason our model is not recognised and not automatically downloaded by the gr00t server. WAR and downloading it manually for a quick POC:

```bash
pip install huggingface_hub
hf download \
  --revision gn1_6 \
  nvidia/GN1x-Tuned-Arena-G1-Loco-Manipulation \
  --local-dir ~/models/checkpoint-20000
```

### Terminal 1: Start the GR00T Policy Server

```bash
cd submodules/Isaac-GR00T
uv run python gr00t/eval/run_gr00t_server.py \
  --modality-config-path ../../isaaclab_arena_gr00t/embodiments/g1/g1_sim_wbc_data_config.py
  --model-path ${MODEL_PATH} \
  --embodiment-tag NEW_EMBODIMENT \
  --device cuda --host 0.0.0.0 --port 5555
```

Wait for: `Server is ready and listening on tcp://0.0.0.0:5555`

### Terminal 2: Run Arena Evaluation

Inside the Arena container (`./docker/run_docker.sh`):

```bash
/isaac-sim/python.sh isaaclab_arena/evaluation/policy_runner.py \
  --viz kit \
  --policy_type isaaclab_arena_gr00t.policy.gr00t_remote_closedloop_policy.Gr00tRemoteClosedloopPolicy \
  --policy_config_yaml_path isaaclab_arena_gr00t/policy/config/g1_locomanip_gr00t_closedloop_config.yaml \
  --remote_host <GROOT_SERVER_IP> --remote_port 5555 \
  --num_steps 1500 \
  --enable_cameras \
  galileo_g1_locomanip_pick_and_place \
  --object brown_box \
  --embodiment g1_wbc_joint
```

### Terminal 1: Start the GR00T Policy Server

```bash
cd submodules/Isaac-GR00T
uv run python gr00t/eval/run_gr00t_server.py \
  --modality-config-path ../../isaaclab_arena_gr00t/embodiments/gr1/gr1_arms_only_data_config.py \
  --model-path ${MODEL_PATH} \
  --embodiment-tag GR1 \
  --device cuda --host 0.0.0.0 --port 5555
```

Wait for: `Server is ready and listening on tcp://0.0.0.0:5555`

### Terminal 2: Run Arena Evaluation

Inside the Arena container (`./docker/run_docker.sh`):

```bash
/isaac-sim/python.sh isaaclab_arena/evaluation/policy_runner.py \
  --viz kit \
  --policy_type isaaclab_arena_gr00t.policy.gr00t_remote_closedloop_policy.Gr00tRemoteClosedloopPolicy \
  --policy_config_yaml_path isaaclab_arena_gr00t/policy/config/gr1_manip_ranch_bottle_gr00t_closedloop_config.yaml \
  --remote_host <GROOT_SERVER_IP> --remote_port 5555 \
  --num_steps 2000 \
  --enable_cameras \
   put_item_in_fridge_and_close_door \
     --embodiment gr1_joint \
     --object ranch_dressing_hope_robolab
```

The Arena container has **no GR00T model dependencies**. It only uses GR00T's lightweight `PolicyClient` (ZMQ + msgpack) to send observations and receive actions. All model inference happens in the GR00T container.

### CODE REVIEW

**Comment by xyao-nv** [comment]
I'm sorry I'd prefer if we do MRs in small chunks. Claude is good at generating +++ lines of codes, and for this refactoring, I think we shall be careful how it was written by Claude.

**Review by xyao-nv — CHANGES_REQUESTED** [review]
Let's break down into smaller MRs once we agree on the design.

**Comment by cvolkcvolk** [comment]
@xyao-nv yes this is just prototyping. Most test will likely fail. Not meant for review.

File: isaaclab_arena/evaluation/policy_runner.py [hunk 1]
```diff
@@ -64,10 +65,43 @@ def rollout_policy(
...
 
+    _TIMING_PRINT_INTERVAL = 100
+
     pbar = None
+    timing = TimingStats()
+
+    # Per-env reset counter — initialised lazily once num_envs is known.
+    per_env_reset_counts: torch.Tensor | None = None
+
+    def _parallel_env_report(num_steps_done: int) -> str:
```
Comment (cvolkcvolk) [inline isaaclab_arena/evaluation/policy_runner.py:76]: Note to remove once ready for review

File: isaaclab_arena/evaluation/policy_runner.py [hunk 2]
```diff
@@ -117,6 +164,12 @@ def rollout_policy(
         raise RuntimeError(f"Error rolling out policy: {e}")
 
     else:
+        print(timing.summary("Runner (final)"))
```
Comment (cvolkcvolk) [inline isaaclab_arena/evaluation/policy_runner.py:167]: Note to remove all timing related code once ready for review

File: isaaclab_arena/evaluation/policy_runner.py [hunk 3]
```diff
@@ -204,18 +257,10 @@ def main():
...
-        if policy.is_remote:
-            policy.shutdown_remote(kill_server=args_cli.remote_kill_on_exit)
-
         # Close the environment.
         env.close()
 
 
 if __name__ == "__main__":
-    # TODO(xinjie.yao, 2026.03.31): Remove it after policy sever-client is implemented properly in v0.3.
     ensure_groot_deps_in_path()
```
Comment (cvolkcvolk) [inline isaaclab_arena/evaluation/policy_runner.py:265]: Note to add a TODO() to adress in a follow up. This should not be needed anymore to handle gr00t deps in our core policy runner

File: isaaclab_arena/policy/action_chunking.py [hunk 4]
```diff
@@ -79,10 +93,81 @@ def get_action(self, fetch_chunk_fn: Callable[[], torch.Tensor]) -> torch.Tensor
...
+        exhausted = (~waiting) & (self.current_action_index >= self.action_chunk_length)
+        self.current_action_chunk[exhausted] = 0.0
+        self.current_action_index[exhausted] = -1
+        self.env_requires_new_chunk[exhausted] = True
+
+        return action
+
+
+# Backwards-compatibility alias
+ActionChunkingState = ActionChunkScheduler
```
Comment (cvolkcvolk) [inline isaaclab_arena/policy/action_chunking.py:173]: Do we need this? I suppose we can delete it

File: isaaclab_arena_gr00t/policy/gr00t_closedloop_policy.py [hunk 5]
```diff
@@ -14,9 +14,15 @@
 from typing import Any
 
 from gr00t.data.embodiment_tags import EmbodimentTag
+# NOTE: Gr00tPolicy is a heavy import (transformers, model loading). This local policy
+# loads the model in-process and requires the full GR00T ML stack. For production
+# evaluation, use Gr00tRemoteClosedloopPolicy which delegates inference to a remote
+# GR00T server and has no heavy dependencies. This local policy may be removed in a
```
Comment (cvolkcvolk) [inline isaaclab_arena_gr00t/policy/gr00t_closedloop_policy.py:20]: Note to check wether we can delete this file. Its loading the GR00T model in-process and requires the full GR00T stack (eg the gr00t container). Which is what we are getting rid of in this MR.

Comment (cvolkcvolk) [reply isaaclab_arena_gr00t/policy/gr00t_closedloop_policy.py:20]: `Gr00tRemoteClosedloopPolicy` will be the only GR00T policy in Arena

File: isaaclab_arena_gr00t/policy/gr00t_remote_closedloop_policy.py [hunk 6]
```diff
@@ -0,0 +1,348 @@
+# Copyright (c) 2025-2026, The Isaac Lab Arena Project Developers (https://github.com/isaac-sim/IsaacLab-Arena/blob/main/CONTRIBUTORS.md).
+# All rights reserved.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+"""GR00T remote closed-loop policy using GR00T's native PolicyClient.
+
+This policy connects to a GR00T policy server (launched via
+``gr00t/eval/run_gr00t_server.py``) and reuses the same observation/action
+translation pipeline as the local ``Gr00tClosedloopPolicy``.
```
Comment (cvolkcvolk) [inline isaaclab_arena_gr00t/policy/gr00t_remote_closedloop_policy.py:10]: Delete this comment

File: isaaclab_arena_gr00t/policy/gr00t_remote_closedloop_policy.py [hunk 7]
```diff
@@ -0,0 +1,348 @@
...
+)
+from isaaclab_arena_gr00t.utils.io_utils import (
+    create_config_from_yaml,
+    load_config_from_yaml,
+    load_gr00t_modality_config_from_file,
+)
+
+
+def _load_config_skip_model_path(yaml_path: str) -> Gr00tClosedloopPolicyConfig:
+    """Load Gr00tClosedloopPolicyConfig but skip model_path validation.
```
Comment (cvolkcvolk) [inline isaaclab_arena_gr00t/policy/gr00t_remote_closedloop_policy.py:46]: This is a WAR. We should get rid of the PolicyConfig model_path validation. 
Note to either add a TODO or to directly adress it here.

File: isaaclab_arena_gr00t/policy/gr00t_remote_closedloop_policy.py [hunk 8]
```diff
@@ -0,0 +1,348 @@
...
+    We replace it with a dummy HuggingFace-style ID so __post_init__
+    validation passes without requiring the file on disk.
+    """
+    data = load_config_from_yaml(yaml_path, Gr00tClosedloopPolicyConfig)
+    data["model_path"] = "remote/server-side-model"
+    return Gr00tClosedloopPolicyConfig(**data)
+
+
+@dataclass
+class Gr00tRemoteClosedloopPolicyArgs(Gr00tBasePolicyArgs):
```
Comment (cvolkcvolk) [inline isaaclab_arena_gr00t/policy/gr00t_remote_closedloop_policy.py:59]: Note that we might want to follow up to clean up the Gr00tArgs in general

File: isaaclab_arena_gr00t/policy/gr00t_remote_closedloop_policy.py [hunk 9]
```diff
@@ -0,0 +1,348 @@
...
+    name = "gr00t_remote_closedloop"
+    config_class = Gr00tRemoteClosedloopPolicyArgs
+
+    def __init__(self, config: Gr00tRemoteClosedloopPolicyArgs, action_scheduler: ActionScheduler | None = None):
+        super().__init__(config)
+
+        # Policy config (for obs/action translation — no model loading).
+        # The YAML may contain a model_path for the local policy, but the remote
+        # wrapper doesn't load any model — the server does. Override model_path
+        # to skip the local-filesystem validation in __post_init__.
```
Comment (cvolkcvolk) [inline isaaclab_arena_gr00t/policy/gr00t_remote_closedloop_policy.py:124]: Note to fix this. Probably we can just remove the model_path from the config file

File: isaaclab_arena/policy/action_chunking.py
Comment (cvolkcvolk) [inline isaaclab_arena/policy/action_chunking.py:1]: We could move the scheduler related fiels into a new directory `policy/action_scheduling`:

```
  isaaclab_arena/policy/
    action_scheduling/    
      action_scheduler.py   
      action_chunk_scheduler.py
      synced_batch_action_scheduler.py  --> This one could deserve its own file

```

File: isaaclab_arena_gr00t/policy/gr00t_remote_closedloop_policy.py [hunk 10]
```diff
@@ -0,0 +1,348 @@
...
+            remote_api_token=getattr(args, "remote_api_token", None),
+        )
+
+
+class Gr00tRemoteClosedloopPolicy(PolicyBase):
+    """GR00T closed-loop policy that delegates inference to a remote GR00T server.
+
+    Uses GR00T's native ``PolicyClient`` (from ``gr00t.policy.server_client``)
+    to communicate with a GR00T policy server. The observation/action translation
+    pipeline is identical to the local ``Gr00tClosedloopPolicy``.
```
Comment (cvolkcvolk) [inline isaaclab_arena_gr00t/policy/gr00t_remote_closedloop_policy.py:89]: Note to remove this comment (Gr00tClosedLoopPolicy would not exist anymore)

File: isaaclab_arena_gr00t/policy/gr00t_remote_closedloop_policy.py [hunk 11]
```diff
@@ -0,0 +1,348 @@
...
+    ) -> torch.Tensor:
+        """Get an action chunk from the remote GR00T server.
+
+        Same pipeline as Gr00tClosedloopPolicy.get_action_chunk(), but calls
+        GR00T's PolicyClient instead of a local Gr00tPolicy.
+        """
+        if isinstance(camera_names, str):
+            camera_names = [camera_names]
+
+        # 1. Reuse the same obs translation as local policy
```
Comment (cvolkcvolk) [inline isaaclab_arena_gr00t/policy/gr00t_remote_closedloop_policy.py:257]: Remove. No local policy anymore.

File: isaaclab_arena_gr00t/policy/gr00t_remote_closedloop_policy.py [hunk 12]
```diff
@@ -0,0 +1,348 @@
...
+                robot_state_joints_config=self.robot_state_joints_config,
+                policy_joints_config=self.policy_joints_config,
+                modality_configs=self.modality_configs,
+            )
+
+        # 2. Call GR00T's own client
+        with self._timing.measure("inference_wait"):
+            robot_action_policy, _ = self._client.get_action(policy_observations)
+
+        # 3. Reuse the same action translation as local policy
```
Comment (cvolkcvolk) [inline isaaclab_arena_gr00t/policy/gr00t_remote_closedloop_policy.py:279]: Same, comment refers to old code

File: isaaclab_arena_gr00t/utils/io_utils.py [hunk 13]
```diff
@@ -204,13 +204,22 @@ def load_gr00t_modality_config_from_file(modality_config_path: str | Path, embod
...
+    import sys
+
     from gr00t.configs.data.embodiment_configs import MODALITY_CONFIGS
     from gr00t.data.embodiment_tags import EmbodimentTag
-    from gr00t.experiment.launch_finetune import load_modality_config
 
     if modality_config_path:
-        # Import module for side-effect registration
-        load_modality_config(modality_config_path)
+        # Import the modality config module for side-effect registration.
```
Comment (cvolkcvolk) [inline isaaclab_arena_gr00t/utils/io_utils.py:214]: Are those changes required for the Policy-Server refactor?

**Comment by xyao-nv** [comment]
Close as refactor is mostly done
